### PR TITLE
Fix saving behaviors that are in subdirectories.

### PR DESCRIPTION
**Pull request by calvertdw**
Currently, behaviors save to the behaviorTrees root directory even if they are in a subfolder. This fixes that by storing the subdirectory as part of the node name, which matches with the .json:

### CODE REVIEW

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeFileLoader.java [hunk 1]
```diff
@@ -34,15 +39,17 @@ public T loadFromFile(WorkspaceResourceFile fileToLoad, BehaviorTreeTopologyOper
...
+            JSONFileTools.load(fileToLoad, jsonNode -> loadedNode.setValue(loadFromFile(relativePath, jsonNode, null, topologyOperationQueue)));
          }
          else
          {
             URL classpathResource = fileToLoad.getClasspathResource();
             if (classpathResource != null)
             {
                LogTools.info("Loading from resource: {}", classpathResource);
-               JSONFileTools.load(classpathResource, jsonNode -> loadedNode.setValue(loadFromFile(jsonNode, null, topologyOperationQueue)));
+               String relativePath = classpathResource.toString().replaceAll(treeFilesDirectory.getPathNecessaryForClasspathLoading(), "");
```
Comment (calvertdw) [inline ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeFileLoader.java]: Needs testing

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeFileLoader.java [hunk 2]
```diff
@@ -79,15 +93,16 @@ private T loadFromFile(JsonNode jsonNode, T parentNode, BehaviorTreeTopologyOper
...
          if (fileNode == null)
          {
-            loadFromFile(childJsonNode, node, topologyOperationQueue);
+            loadFromFile(filePath, childJsonNode, node, topologyOperationQueue);
          }
          else
          {
             WorkspaceResourceFile childFile = new WorkspaceResourceFile(behaviorTreeState.getSaveFileDirectory(), fileNode.asText());
             LogTools.info("Loading {}", childFile.getFilesystemFile());
+            String relativePath = treeFilesDirectory.getFilesystemDirectory().relativize(childFile.getFilesystemFile()).toString();
```
Comment (calvertdw) [inline ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeFileLoader.java]: Needs testing. Also this should similarly load from classpath if not available from file, which would need testing.

**Review by TomaszTB — APPROVED** [review]
I'm not caught up on this code, but the changes look good

File: ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/behavior/tree/RDXBehaviorTree.java [hunk 3]
```diff
@@ -71,10 +71,10 @@ public RDXBehaviorTree(WorkspaceResourceDirectory treeFilesDirectory,
                                                    panel3D,
                                                    referenceFrameLibrary);
       treeRebuilder = new BehaviorTreeExtensionSubtreeRebuilder(this::getRootNode, crdtInfo);
-      fileMenu = new RDXBehaviorTreeFileMenu(treeFilesDirectory);
```
Comment (calvertdw) [inline ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/behavior/tree/RDXBehaviorTree.java:74]: This parameter was unused so I removed it. Just a minor cleanup thing.

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeFileLoader.java [hunk 4]
```diff
@@ -16,86 +17,119 @@ public class BehaviorTreeFileLoader<T extends BehaviorTreeNodeLayer<T, ?, ?, ?>>
...
+      this.treeFilesDirectory = treeFilesDirectory;
    }
 
-   public T loadFromFile(WorkspaceResourceFile fileToLoad, BehaviorTreeTopologyOperationQueue topologyOperationQueue)
+   public T loadFromFile(WorkspaceResourceFile file, BehaviorTreeTopologyOperationQueue topologyOperationQueue)
+   {
+      return loadFromFile(file, null, null, topologyOperationQueue);
+   }
+
+   private T loadFromFile(WorkspaceResourceFile file, JsonNode jsonNode, T parentNode, BehaviorTreeTopologyOperationQueue topologyOperationQueue)
```
Comment (calvertdw) [inline ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeFileLoader.java:36]: I've restructured this to be a more well-formed recursive function. This allowed to reduce some code duplication and fix some bugs.

**Pull request merged**